package/MDAnalysis/topology/ITPParser.py

```diff
@@ -527,7 +531,23 @@ def parse(self, include_dir='/usr/local/gromacs/share/gromacs/top/',
             self.system_molecules = [x.name for x in self._molecules]
 
         self.build_system()
-        
+
+        self.types = np.array(self.types)
+        self.charges = np.array(self.charges)
+        self.masses = np.array(self.masses)
+
+        if not all(self.charges):
+            empty = self.charges == ''
+            self.charges[empty] = [
+                (self.atomypes.get(x)["charge"] if x in self.atomypes.keys() else '') for x in self.types[empty]
+            ]
+
+        if not all(self.masses):
+            empty = self.masses == ''
+            self.masses[empty] = [
+                (self.atomypes.get(x)["mass"] if x in self.atomypes.keys() else '') for x in self.types[empty]
```

Comment:
If this is too long, you could split it over two lines.

Reply:
Possibly you could even do the below to save time.
```python
try:
	m = self.atomtypes[x]["mass"]
except KeyError:
	m = guessers.guess_mass(....)
```

Not sure if a singular guess_mass function exists